client/src/pages/ProfileListings/hooks/useSitterProfiles.ts

```diff
@@ -0,0 +1,95 @@
+import type { RequestError, ValidationError, OnError } from './../../../interface/ApiData';
+import type { GetSitterProfileParams, SitterProfile } from '../../../interface/Profile';
+import { useEffect, useState, useCallback, useRef } from 'react';
+import { v4 as uuid } from 'uuid';
+import { getSitterProfiles } from '../../../helpers/APICalls/profile';
+import wait from '../../../helpers/wait';
+
+interface Options extends Omit<GetSitterProfileParams, 'page'> {
+  delay?: number;
+  loadOnMount?: boolean;
+}
+
+interface StateItem {
+  id: string;
+  loading: boolean;
+  sitter: SitterProfile | null;
+}
+
+type OnSuccess<R> = (profiles: StateItem[]) => R;
+type MatchProfiles = <R>(onError: OnError<R>, onSuccess: OnSuccess<R>) => R;
+
+// I think we have two different arrays, one to represent the data that is loading and the other to represent that data that has loaded.
```

Comment:
This is not the right place for this comment. Please remove it as it's not needed.

Reply:
Ahahahaha, I always seem to forget to remove those  unecessary comments. This was my thinking originally. But I found a better solution. Comment removed :3